src/Temporalio/Worker/WorkflowWorker.cs

```diff
@@ -215,46 +246,62 @@ private async Task HandleActivationAsync(IPayloadCodec? codec, WorkflowActivatio
                 }
             }
 
-            // Remove from cache if requested
-            if (removeJob != null)
+            // Send completion
+            logger.LogTrace("Sending workflow completion: {Completion}", comp);
+            try
             {
-                logger.LogDebug(
-                    "Evicting workflow with run ID {RunId}, message: {Message}",
-                    act.RunId,
-                    removeJob.Message);
-                runningWorkflows.TryRemove(act.RunId, out _);
-                onEviction?.Invoke(act.RunId, removeJob);
-                // If this is a deadlocked workflow task, we attach remove completion to task
-                // completion
-                if (deadlockedWorkflows.TryRemove(act.RunId, out var task))
-                {
-                    _ = task.ContinueWith(
-                        async _ =>
+                await options.BridgeWorker.CompleteWorkflowActivationAsync(comp).ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                logger.LogError(e, "Failed completing activation on {RunId}", act.RunId);
+            }
+        }
+
+        private async Task HandleCacheEvictionAsync(WorkflowActivation act, RemoveFromCache job)
```

Comment:
When we added data conversion details to the existing `HandleActivationAsync` it complicated the code a lot because eviction was inside it too. So we took this opportunity to refactor and move eviction here to a separate method.

Reply:
The royal "we" here is a bit hilarious lol